index.bs

```diff
@@ -485,23 +485,19 @@ The header's ABNF is:
   Sec-CH-UA = sf-list
 ```
 
-To <dfn abstract-op local-lt="set-ua">return the `Sec-CH-UA` value for a request</dfn>, [=user agents=] MUST:
-
-1.  Let |list| be a [=/list=], initially empty.
-
-2. For each |brandVersion| in [=user agent/brands=]:
+To <dfn abstract-op>return the `Sec-CH-UA` value for a request</dfn>,
+[=user agents=] MUST:
 
+1. Let |list| be a [=/list=], initially empty.
+1. Let |brands| be the result of running the [=create brands=] steps, with
+    [=user agent/significant version=] passed as an argument.
```

Comment:
It's unclear if what is passed is the version itself, or an instruction to operate on that version type. I think you meant the latter, but it might be worthwhile to pass an internal enum, and then refer internally to the right version type based on that enum.

Also, right now both significant version and full version are defined as a single version. It may be worthwhile to adapt their descriptions to indicate that, unlike Highlander, there can be more than one.

Reply:
> it might be worthwhile to pass an internal enum, and then refer internally to the right version type based on that enum

This sounds probably like what we want to do, but I'm not sure how to spec it. Are you aware of any other similar usage in a spec I can learn from, or have some suggestions?

Reply:
@yoavweiss does https://github.com/WICG/ua-client-hints/pull/250/commits/ac953438348d12ea2d7ff07d7a0247badfcc0a60 look better from your perspective?